core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -212,19 +200,22 @@ public CloseableIterable<StructLike> rows() {
         return CloseableIterable.transform(rowIterable, projection::wrap);
 
       } catch (IOException e) {
-        throw new RuntimeIOException(
-            e, "Cannot read manifest list file: %s", manifestListTask.file().path());
+        throw new RuntimeIOException(e, "Cannot read manifest list file: %s", manifestListLocation);
       }
     }
 
     @Override
     public DataFile file() {
-      return manifestListTask.file();
+      return DataFiles.builder(PartitionSpec.unpartitioned())
```

Comment:
Optional: If this is called repeatedly maybe worth making this in the CTOR.

Reply:
+1, would it make sense to store this in a memoized supplier so that `DataFile` only gets initialized when required?

Reply:
I don't think it gets called, but that could change. I'll update to a lazy field.